src/workflow/unlock.h

```diff
@@ -15,15 +15,31 @@
 #ifndef _WORKFLOW_UNLOCK_H_
 #define _WORKFLOW_UNLOCK_H_
 
+#include "workflow.h"
+
 #include <compiler_util.h>
 #include <keystore.h>
 
+/**
+ * Create a workflow that prompts the user for the password and unlocks the keystore, calls
+ * the provided callback, and exits. The workflow will terminate if either the user enters the
+ * correct password, or the device is reset after too many failed attempts. The callback will
```

Comment:
> or the device is reset after too many failed attempts.

In this case the device reboots and there won't be any response, right?

What about if HWW_REQ_CANCEL comes in?

Reply:
If the workflow is stopped, cleanup() will be called. No further action needed (as every active workflow cleans up after itself). Note that atm there is no `abort()` mechanism, so stopping a workflow from "outside" is still out of scope for this PR. That will come with #164. Forcefully aborting a workflow will just destroy (`workflow_stop_workflow()`) every workflow in the stack until the selected one, which should guarantee a proper cleanup no matter the state.

So tl;dr: 
> What about if HWW_REQ_CANCEL comes in?

This is not responsibility of the `workflow` stack :)